python/amici/sbml_import.py

```diff
@@ -1025,14 +1025,13 @@ def _process_rules(self) -> None:
 
             if variable in rulevars:
                 for nested_rule in rules:
-                    nested_formula = sp.sympify(
-                        sbml.formulaToL3String(nested_rule.getMath()),
-                        locals=self.local_symbols)
-                    nested_formula = \
-                        str(nested_formula.subs(variable, formula)).replace('**', '^')
-                    if nested_rule.setFormula(nested_formula) != sbml.LIBSBML_OPERATION_SUCCESS:
-                        raise SBMLException(f'Formula {nested_formula} cannot be parsed by libSBML!')
+                    nested_rule_math_ml = sp.printing.print_mathml(
```

Comment:
we lost the `nested_formula.subs(variable, formula)` here, which is the key thing this part of the code is supposed to do (flatten out nested rules).